src/renderer/shared/ui-kit/Select/Select.tsx

```diff
@@ -0,0 +1,149 @@
+import * as RadixSelect from '@radix-ui/react-select';
+import { type PropsWithChildren, createContext, useContext, useMemo } from 'react';
+
+import { type XOR } from '@/shared/core';
+import { cnTw } from '@/shared/lib/utils';
+import { Icon } from '@/shared/ui';
+import { ScrollArea } from '../ScrollArea/ScrollArea';
+import { Surface } from '../Surface/Surface';
+import { useTheme } from '../Theme/useTheme';
+import { gridSpaceConverter } from '../_helpers/gridSpaceConverter';
+
+type ContextProps = {
+  theme?: 'light' | 'dark';
+  invalid?: boolean;
+  disabled?: boolean;
+  testId?: string;
+};
+
+const Context = createContext<ContextProps>({});
+
+type ControlledSelectProps = {
+  value: string;
+  onChange: (value: string) => void;
+} & XOR<{
+  open: boolean;
+  onToggle: (value: boolean) => void;
+}>;
+
+type RootProps = PropsWithChildren<ControlledSelectProps & ContextProps>;
+
+const Root = ({
+  theme = 'light',
+  invalid,
+  disabled,
+  testId = 'Select',
+  open,
```

Comment:
let's call boolean values for `open` as `isOpen` it's easier to read in components